llvm/lib/Analysis/AliasAnalysis.cpp

```diff
@@ -818,7 +818,7 @@ bool llvm::isNoAliasCall(const Value *V) {
 
 static bool isNoAliasOrByValArgument(const Value *V) {
   if (const Argument *A = dyn_cast<Argument>(V))
-    return A->hasNoAliasAttr() || A->hasByValAttr();
+    return A->hasNoAliasAttr() || A->hasByValAttr() || A->hasDeadOnReturnAttr();
```

Comment:
Doesn't it imply that stores that would only be visible on the unwind path can also be elided?
If so, I don't think `dead_on_return` is a good name. It should be something that implies `dead_on_unwind`.

Reply:
I think we may want to keep dead_on_return orthogonal from dead_on_unwind, and let the memory pointed by be dead upon normal function return. Landing pads may need to inspect parameter memory (not sure if can be the case?). Even so, that would otherwise subsume dead_on_unwind, and I don't think we want this?

Reply:
We also shouldn't check it here, this will imply aliasing properties.